File: civitasJava/src/civitas/CasillaCalle.java
/**
 * @file CasillaImpuesto.java
 * @author Yesenia González Dávila
 * @author Esther García Gallego
 * @brief Casilla de tipo calle. Subclase de Casilla.
 * @note Grupo B.3
*/

package civitas;

import java.util.ArrayList;

public class CasillaCalle extends Casilla {
    
    /** Constructor calle. */
    CasillaCalle (TituloPropiedad titulo) {
        super (titulo.getNombre());         // Nombre de la casilla
        tituloPropiedad = titulo;           // Impuesto a aplicar
        tipoCasilla = "CALLE";
    }
    
    /** Recibe a un jugador en la casilla, gestionando la operación 
     * correspondiente (comprar o alquiler según el propietario). */
    @Override
    void recibeJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto (actual, todos)) {
            informe (actual, todos);
            Jugador jugador = todos.get(actual);

            if (!tituloPropiedad.tienePropietario()) 
                jugador.puedeComprarCasilla();
            else if (!tituloPropiedad.getHipotecado())
                tituloPropiedad.tramitarAlquiler(jugador);
        }
    }
}

File: civitasRuby/pkg/civitasRuby-0.0.1/lib/civitas/casilla_calle.rb
#encoding:utf-8

=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

require_relative './TituloPropiedad.rb'


module Civitas
  class CasillaCalle < Casilla
    
    def initialize(titulo)
      super(titulo.nombre)
      @tituloPropiedad = titulo
      @importe = -1
      @Carcel = -1
      @mazo = nil
      @sorpresa = nil
    end
    
    def recibeJugador(actual, todos)
      
      if(jugadorCorrecto(actual, todos))
        informe(actual,todos)
        jugador = todos[actual]
      
        if !@tituloPropiedad.tienePropietario
          jugador.puedeComprarCasilla
        elsif !@tituloPropiedad.hipotecado
          @tituloPropiedad.tramitarAlquiler(jugador)
        else
            Diario.instance.ocurre_evento("El jugador #{jugador.nombre} se libra de pagar el alquiler porque la calle está hipotecada.")
        end
      
      end
    end
    
    def informe (actual, todos)
      str = "El jugador #{todos[actual].nombre} ha caido en una casilla de tipo CALLE"
      Diario.instance.ocurre_evento(str)
    end
    
  end
end
